packages/react/src/components/Authenticator/Router/index.tsx

```diff
@@ -11,15 +11,17 @@ import { ConfirmVerifyUser, VerifyUser } from '../VerifyUser';
 import { ConfirmSignIn } from '../ConfirmSignIn/ConfirmSignIn';
 import { ConfirmResetPassword, ResetPassword } from '../ResetPassword';
 
+export type AuthenticatorRenderProp = ({
+  signOut,
+  user,
+}: {
+  signOut: ReturnType<typeof useAuthenticator>['signOut'];
+  user: CognitoUserAmplify;
+}) => JSX.Element;
+
 export type RouterProps = {
   className?: string;
-  children?: ({
-    signOut,
-    user,
-  }: {
-    signOut: ReturnType<typeof useAuthenticator>['signOut'];
-    user: CognitoUserAmplify;
-  }) => JSX.Element;
+  children?: JSX.Element | AuthenticatorRenderProp;
```

Comment:
`JSX.Element` alone does not work as a function signature for a React component, recommend `<T>(props: T) => JSX.Element`

Reply:
Please disregard the above comment. The recommendation is basically a generic version of the current typing, and `ReactNode` is the commonly used type for `children` props

Reply:
Based on our offline discussion and your evaluation above, we're typing this to 

```
type AuthenticatorChildren =
  | React.ReactNode
  | (({
      signOut,
      user,
    }: {
      signOut?: ReturnType<typeof useAuthenticator>['signOut'];
      user?: CognitoUserAmplify;
    }) => React.ReactNode);
```

Reply:
Mb, didn't mean to resolve early. This change is made in https://github.com/aws-amplify/amplify-ui/pull/1685/commits/f3cb144b730ab28175fa0f7ba80b4252d9ced6a1.